src/Widgets/Username.vala

```diff
@@ -46,24 +51,92 @@ public class Username : Gtk.Box {
     }
 
     public bool is_ready () {
+        string username = username_entry.get_text ();
         return realname_entry.get_text_length () != 0
-            && username_entry.get_text_length () != 0;
+            && username_entry.get_text_length () != 0
+            && str_contains_alpha(username)
+            && !is_forbidden(username);
     }
 
     private string validate (string input) {
         var text = new StringBuilder ();
 
-        int i = 0;
-        char c = input[i];
-        while (c != '\0') {
-            char cl = c.tolower ();
-            if (cl.isalnum () || cl == '_') {
-                text.append_c (cl);
+        for (int i = 0; i < input.length; i++) {
+            char c = input[i];
+
+            // The first char must be alphabetic.
+            // The following may be alphanumeric, and '_', '.', or '-'.
+            bool append = text.str.length > 1
```

Comment:
It seems to me it would be better to show an error about having invalid characters rather than just silently dropping them.

Reply:
Should we stop populating the username from the real name then?

Reply:
I guess I should see what this actually looks like. It may be ok.

Reply:
Actually the way it works looks great.